sa/sa.go

```diff
@@ -1768,47 +1886,75 @@ func (ssa *SQLStorageAuthority) statusForOrder(ctx context.Context, order *corep
 			"authorizations", *order.Id)
 }
 
-func (ssa *SQLStorageAuthority) getAllOrderAuthorizations(
+type authzStatus struct {
```

Comment:
We arrived on this after conversation with @cpu, relying on full `core.Authorization`s makes all of this logic considerably more complicated without actually providing any benefit. Perhaps it could just use a different name?

Reply:
Yeah, what about validityInfo? Also we should change the function names since we're no longer getting "authorizations."

BTW, is the complexity you mention code complexity or extra database fetches? Can the code complexity not be handled by having helper functions?

Reply:
> Yeah, what about validityInfo? Also we should change the function names since we're no longer getting "authorizations."

Works for me.

> BTW, is the complexity you mention code complexity or extra database fetches? Can the code complexity not be handled by having helper functions?

It's both. The main complexity comes from the fact that the SA code is designed to never actually operate on `core.Authorization`s itself, it only deals in `corepb.Authorization`s and `authz2Model`s. In order to use `core.Authorization` here we'd need a method for converting `authz2Model`s or do something gross like `PBToAuthz(modelToPB(authz2Model))` which is a bunch of extra complexity in order to just look at the status/expires fields.

When initially working on this code I started down this path but coalescing on `core.Authorization`s actually ends up making the impl much more confusing and hard to understand.

Reply:
What about coalescing on `authz2Model` for internal representations?

Reply:
Representing old authorizations using the `authz2Model` format would be pretty complicated, and all of the extra code needed to do that would end up just getting nuked when we get rid of old style authorizations entirely. I'm pretty convinced the current solution is the best at the moment.

Reply:
Sounds good :+1: